packages/node/src/indexer/mmr.service.ts

```diff
@@ -75,18 +92,17 @@ export class MmrService implements OnApplicationShutdown {
         `Append Mmr failed, input data length should be ${DEFAULT_WORD_SIZE}`,
       );
     }
-    const blockLeafHeight = poiBlock.id - this.blockOffset - 1; //leaf index alway -1
-    const mmrNextLeafIndex = await this.fileBasedMmr.getLeafLength();
-    if (blockLeafHeight === mmrNextLeafIndex) {
-      //next mmr append leaf index should be same as block height
-      await this.fileBasedMmr.append(newLeaf, blockLeafHeight);
-      mmrRoot = await this.fileBasedMmr.getRoot(blockLeafHeight);
-    } else if (blockLeafHeight < mmrNextLeafIndex) {
+    const blockHeight = poiBlock.id - this.blockOffset;
+
+    if (blockHeight === this.nextMmrBlockHeight) {
+      await this.fileBasedMmr.append(newLeaf, blockHeight - 1);
+      mmrRoot = await this.fileBasedMmr.getRoot(blockHeight - 1);
+    } else if (blockHeight < this.nextMmrBlockHeight) {
```

Comment:
the blockHeight here is actually a leafIndex, so it shouldn't be used to compare with a blockHeight (includes offset)